lib/sensors/src/UltraSonicSensor.cpp

```diff
@@ -0,0 +1,33 @@
+#include "UltraSonicSensor.h"
+
+UltraSonicSencor::UltraSonicSensor(PinName echoPin, PinName trigPin) : _echoPin(echoPin), _trigPin(trigPin) {}
+
+uint64_t UltraSonicSensor::pulseIn() {
+  uint64_t _time{};
+  // instantiate PwmIn class
+  PwmIn _pulse(_echopin, NUM_SAMPLES);
+  // convert to microseconds
+  _time = _pulse.pulseWidth() * CONV_FACTOR;
+  return _time;
+}
+
+float UltraSonicSensor::getDistance() {
```

Comment:
Add a comment with a link to the official docs for the sensor and also a short comment on how this sensor works(ie. what sequence of things is this function supposed to do to the sensor to get a reading)